### DISCUSSION TITLE
How to use web service answer in tasmota script ?

### DISCUSSION THREAD

**Discussion by ygelt** [arendst/Tasmota#19849] [Q&A] [answered]
Hi, I compiled tasmota 13.2.0.1 with the following add-on in user_config_override.h :
   **#define USE_SCRIPT
   #define USE_WEBSEND_RESPONSE
   #undef USE_RULES**

In my script, I call a web service returning a JSON with :
**WebQuery http://xxxx/connect/v1/device?id=test GET**
Or
**WebSend [xxxx] /connect/v1/device?id=test**

It seems to work in both cases since the >E section is triggered and the log returns the right web service answer :
   **11:53:00.357 RSL: RESULT = {"WebQuery":{ "power1":"ON", "power2":"OFF", "device":"test" }}
   11:53:00.387 RSL: RESULT = {"WebQuery":"Done"}**

But how can I work with the JSON response in the >E section ? or another section ?
Which %variable% or which [json#] contains the answer values which can then be tested ?

I tried many many combinations using "print %xxx%" and "print [xxx]" in >E section during 1 full day, searching on forums and helped by chatGPT but no solution is working :-( Could you please help me ?
Yves.

**Comment by pkkrusty** [comment]
In the >E section, assign each JSON piece to a variable defined in the >D section. I.e.

```
>D
length=0
temp=0

>E
length=VL53X1#Length
temp=DS18B20#Temperature
```
Then do whatever you want with those variables.

**Reply by pkkrusty** [reply]
In your case 

state=""

Then 

state=Webquery#Power1

**Reply by ygelt** [reply]
Thanks for your answer ... it's better but my results look strange:

I have this minimum code:
>D
state=""

>BS
WebQuery http://xxxx/connect/v1/device?id=test GET
print CALLED --->

>E
state=Webquery#power1
print --> %state% <--

I started from console "WebQuery http://xxxx/connect/v1/device?id=test GET" but get a bad result.

15:11:03.324 CMD: WebQuery http://xxxx/connect/v1/device?id=test GET
15:11:03.469 RSL: RESULT = {"WebQuery":{ "power1":"ON", "power2":"OFF", "device":"test" }}
15:11:03.472 --> : RESULT = {"WebQue <--
15:11:03.476 RSL: RESULT = {"WebQuery":"Done"}
15:11:03.478 --> : RESULT = {"WebQue <-- 

Do you have an idea why I receive "RESULT = {"WebQue" as value ?

**Reply by pkkrusty** [reply, marked as answer]
D should be >D and each other section should have `>` also. After D, put an increased variable length, like `>D 48` to make sure you  can store larger strings. In the E section, only assign variable. Then in F or S section do operations on those variables and print.

```
    >D 48
    state=""

    >BS
    WebQuery http://xxxx/connect/v1/device?id=test GET
    print CALLED --->

    >E
    state=WebQuery#power1
>S
print %state%
```

Note that MQTT messages and JSON is case sensitive, so need to be careful when assigning variables in >E section.

**Reply by ygelt** [reply]
Hi, thanks a lot for your answer. With >D 48 and minimal processing (Only variables set by the JSON return from web server) in >E, it solves the issue :-)
